Provide a pointer to the correct names for Boolean types

Given the following code: https://play.rust-lang.org/?version=stable&mode=debug&edition=2021&gist=4e7d539241dfe92a24817728751e5523

```rust
fn main() {
    let value: Boolean = true;
}
```

The current output is:

```
error[E0412]: cannot find type `Boolean` in this scope
 --> src/main.rs:2:16
  |
2 |     let value: Boolean = true;
  |                ^^^^^^^ not found in this scope

For more information about this error, try `rustc --explain E0412`.
error: could not compile `example` due to previous error
```

<!-- The following is not always necessary. -->
Ideally the output should look like:

```
error[E0412]: cannot find type `Boolean ` in this scope
    --> src/main.rs:4:16
     |
4    |       let value: Boolean = true;
     |                  ^^^^^^^ help: a type with a similar meaning exists: `bool`
```

I stumbled here when returning to Rust after a while writing code in non-Rust languages; I'm sure I'm not the only person who's gotten the name of Rust's boolean type wrong. It's not hard to work out the correct behaviour, but I thought it would be nice if Rust could help programmers coming from other languages.

(I don't feel strongly about this suggestion and won't mind if it's closed as WONTFIX, but I know Esteban would be sad if he knew I'd had an idea for a diagnostic improvement that I didn't tell anyone.)

<!-- TRIAGEBOT_START -->

<!-- TRIAGEBOT_ASSIGN_START -->

<!-- TRIAGEBOT_ASSIGN_DATA_START$${"user":"lyming2007"}$$TRIAGEBOT_ASSIGN_DATA_END -->

<!-- TRIAGEBOT_ASSIGN_END -->
<!-- TRIAGEBOT_END -->

if we change the code to 
```
fn main() {
    let value: boolean = true;
}
```
the error will be like:
```
error[E0412]: cannot find type `boolean` in this scope
 --> boolean.rs:2:14
  |
2 |   let value: boolean = true;
  |              ^^^^^^^
  |              |
  |              not found in this scope
  |              help: perhaps you intended to use this type: `bool`

error: aborting due to previous error
```
How about we make the error message for your case like:
Ideally the output should look like:

```
error[E0412]: cannot find type `Boolean` in this scope
 --> Boolean.rs:2:14
  |
2 |   let value: Boolean = true;
  |              ^^^^^^^
  |              |
  |              not found in this scope
  |              help: perhaps you intended to use this type: `bool`

error: aborting due to previous error
```

--- a/compiler/rustc_resolve/src/late/diagnostics.rs
+++ b/compiler/rustc_resolve/src/late/diagnostics.rs
@@ -1508,6 +1508,8 @@ impl<'a: 'ast, 'ast> LateResolutionVisitor<'a, '_, 'ast> {
         Some(match name {
             "byte" => sym::u8, // In Java, bytes are signed, but in practice one almost always wants unsigned bytes.
             "short" => sym::i16,
+            "Bool" => sym::bool,
+            "Boolean" => sym::bool,
             "boolean" => sym::bool,
             "int" => sym::i32,
             "long" => sym::i64,

--- a/src/test/ui/lint/recommend-literal.rs
+++ b/src/test/ui/lint/recommend-literal.rs
@@ -7,6 +7,13 @@ fn main() {
     let y: long = 74802374902374923;
     //~^ ERROR cannot find type `long` in this scope
     //~| HELP perhaps you intended to use this type
+    let v1: Boolean = true;
+    //~^ ERROR: cannot find type `Boolean` in this scope [E0412]
+    //~| HELP perhaps you intended to use this type
+    let v2: Bool = true;
+    //~^ ERROR: cannot find type `Bool` in this scope [E0412]
+    //~| HELP a builtin type with a similar name exists
+    //~| HELP perhaps you intended to use this type
 }
 
 fn z(a: boolean) {

--- a/src/test/ui/lint/recommend-literal.stderr
+++ b/src/test/ui/lint/recommend-literal.stderr
@@ -16,8 +16,32 @@ LL |     let y: long = 74802374902374923;
    |            not found in this scope
    |            help: perhaps you intended to use this type: `i64`
 
+error[E0412]: cannot find type `Boolean` in this scope
+  --> $DIR/recommend-literal.rs:10:13
+   |
+LL |     let v1: Boolean = true;
+   |             ^^^^^^^
+   |             |
+   |             not found in this scope
+   |             help: perhaps you intended to use this type: `bool`
+
+error[E0412]: cannot find type `Bool` in this scope
+  --> $DIR/recommend-literal.rs:13:13
+   |
+LL |     let v2: Bool = true;
+   |             ^^^^
+   |
+help: a builtin type with a similar name exists
+   |
+LL |     let v2: bool = true;
+   |             ~~~~
+help: perhaps you intended to use this type
+   |
+LL |     let v2: bool = true;
+   |             ~~~~
+
 error[E0412]: cannot find type `boolean` in this scope
-  --> $DIR/recommend-literal.rs:12:9
+  --> $DIR/recommend-literal.rs:19:9
    |
 LL | fn z(a: boolean) {
    |         ^^^^^^^
@@ -26,7 +50,7 @@ LL | fn z(a: boolean) {
    |         help: perhaps you intended to use this type: `bool`
 
 error[E0412]: cannot find type `byte` in this scope
-  --> $DIR/recommend-literal.rs:17:11
+  --> $DIR/recommend-literal.rs:24:11
    |
 LL | fn a() -> byte {
    |           ^^^^
@@ -35,7 +59,7 @@ LL | fn a() -> byte {
    |           help: perhaps you intended to use this type: `u8`
 
 error[E0412]: cannot find type `float` in this scope
-  --> $DIR/recommend-literal.rs:24:12
+  --> $DIR/recommend-literal.rs:31:12
    |
 LL |     width: float,
    |            ^^^^^
@@ -44,7 +68,7 @@ LL |     width: float,
    |            help: perhaps you intended to use this type: `f32`
 
 error[E0412]: cannot find type `int` in this scope
-  --> $DIR/recommend-literal.rs:27:19
+  --> $DIR/recommend-literal.rs:34:19
    |
 LL |     depth: Option<int>,
    |                   ^^^ not found in this scope
@@ -59,7 +83,7 @@ LL | struct Data<int> {
    |            +++++
 
 error[E0412]: cannot find type `short` in this scope
-  --> $DIR/recommend-literal.rs:33:16
+  --> $DIR/recommend-literal.rs:40:16
    |
 LL | impl Stuff for short {}
    |                ^^^^^
@@ -67,6 +91,6 @@ LL | impl Stuff for short {}
    |                not found in this scope
    |                help: perhaps you intended to use this type: `i16`
 
-error: aborting due to 7 previous errors
+error: aborting due to 9 previous errors
 
 For more information about this error, try `rustc --explain E0412`.

